application/src/main/java/com/entando/hub/catalog/service/BundleGroupService.java

```diff
@@ -120,54 +124,36 @@ public void updateMappedBy(BundleGroup toUpdate, BundleGroupController.BundleGro
             }).collect(Collectors.toSet());
             toUpdate.setCategories(categorySet);
         }
-        if (bundleGroup.getChildren() != null) {
-            //TODO native query to improve performance
-            bundleRepository.findByBundleGroupsIs(toUpdate).stream().forEach(bundle -> {
-                bundle.getBundleGroups().remove(toUpdate);
-                bundleRepository.save(bundle);
-            });
-            Set<Bundle> bundleSet = bundleGroup.getChildren().stream().map((bundleChildId) -> {
-                com.entando.hub.catalog.persistence.entity.Bundle bundle = bundleRepository.findById(Long.valueOf(bundleChildId)).get();
-                bundle.getBundleGroups().add(toUpdate);
-                bundleRepository.save(bundle);
-                return bundle;
-            }).collect(Collectors.toSet());
-            toUpdate.setBundles(bundleSet);
+	    if (bundleGroup.getChildren() != null && Objects.nonNull(bundleGroup.getVersionDetails())) {
+	    	bundleGroup.getVersionDetails().setChildren(bundleGroup.getChildren());
+	    }
+
+        if (bundleGroup.getVersionDetails() != null) {
+        	 Optional<String> optBundleGroupVersionId =  Objects.nonNull(bundleGroup.getVersionDetails().getBundleGroupVersionId()) 
+        			 ?  Optional.of(bundleGroup.getVersionDetails().getBundleGroupVersionId())
+        					 : Optional.empty();
+        	 logger.debug("{}: updateMappedBy: bundle group version id: {}", CLASS_NAME, optBundleGroupVersionId);
+        	 bundleGroupVersionService.createBundleGroupVersion(bundleGroup.getVersionDetails().createEntity(optBundleGroupVersionId, toUpdate), bundleGroup.getVersionDetails());
         }
-
     }
 
     @Transactional
     public void deleteBundleGroup(String bundleGroupId) {
+    	logger.debug("{}: deleteBundleGroup: Delete a bundle group by id: {}", CLASS_NAME, bundleGroupId);
         Long id = Long.valueOf(bundleGroupId);
-        Optional<BundleGroup> byId = bundleGroupRepository.findById(id);
+        Optional<BundleGroup> byId = bundleGroupRepository.findById(id); //No need to fetch here from db, can be passed from controller
         byId.ifPresent(bundleGroup -> {
             deleteFromCategories(bundleGroup);
-            deleteFromBundles(bundleGroup);
+//            TODO: Delete the bundle group from Bundle Group Versions
```

Comment:
@SACHIN-BAGHEL is there a ticket for this or is it leftover code?

Reply:
@nshaw We were planing to use this code in EHUB-175 (data leak). Give us some time we will use/remove very soon,we are about to finish EHUB-175. Although this method is not in use.

Reply:
@AkhileshPrajapatiNV , all good, thanks.